bodhi/tests/server/consumers/test_greenwave.py

```diff
@@ -0,0 +1,102 @@
+# Copyright © 2019 Red Hat, Inc.
+#
+# This file is part of Bodhi.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""This test suite contains tests for the bodhi.server.consumers.greenwave module."""
+
+from unittest import mock
+
+from fedora_messaging.api import Message
+
+from bodhi.server import models
+from bodhi.server.consumers import greenwave
+from bodhi.server.config import config
+from bodhi.server.exceptions import BodhiException
+from bodhi.tests.server.base import BaseTestCase, TransactionalSessionMaker
+
+
+class TestGreenwaveHandler(BaseTestCase):
+    """Test class for the :func:`GreenwaveHandler` method."""
+
+    def setUp(self):
+        super().setUp()
+        self.sample_message = Message(
+            topic="org.fedoraproject.prod.greenwave.decision.update",
+            body={"subject_identifier": "bodhi-2.0-1.fc17"},
+        )
+        self.handler = greenwave.GreenwaveHandler()
+
+    @mock.patch.dict(config, [('greenwave_api_url', 'http://domain.local')])
+    def test_update_test_gating_status(self):
+        """
+        Assert that a greenwave message updates the gating tests status of an update.
+        """
+
+        self.handler.db_factory = TransactionalSessionMaker(self.Session)
```

Comment:
Just an idea, this initialization of the test DB session could be moved to `TestGreenwaveHandler.setUp()` to be available in all test methods that might use it.